File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task3_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <string>
#include <algorithm>
#include <cctype>

bool isPangram(std::string str) {
    std::transform(str.begin(), str.end(), str.begin(), ::tolower);
    for (char c = 'a'; c <= 'z'; c++) {
        if (str.find(c) == std::string::npos) {
            return false;
        }
    }
    return true;
}

int main() {
    std::string test = "The quick brown fox jumps over the lazy dog";
    std::cout << std::boolalpha << isPangram(test) << std::endl;
    return 0;
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task3_CLAUDE_claude-3-5-sonnet-20240620.java

import java.util.HashSet;
import java.util.Set;

public class PangramChecker {
    public static boolean isPangram(String sentence) {
        Set<Character> letters = new HashSet<>();
        for (char c : sentence.toLowerCase().toCharArray()) {
            if (Character.isLetter(c)) {
                letters.add(c);
            }
        }
        return letters.size() == 26;
    }

    public static void main(String[] args) {
        String test1 = "The quick brown fox jumps over the lazy dog";
        String test2 = "This is not a pangram";
        System.out.println(isPangram(test1));  // true
        System.out.println(isPangram(test2));  // false
    }
}
